zebra-consensus/src/verify/block.rs

```diff
@@ -50,6 +50,24 @@ pub(crate) fn node_time_check(
     }
 }
 
+/// [3.10]: https://zips.z.cash/protocol/protocol.pdf#coinbasetransactions
+pub(crate) fn coinbase_check(block: &Block) -> Result<(), Error> {
+    if block.coinbase_height().is_some() {
+        // No additional coinbase transactions allowed
+        if block.transactions.len() > 1 {
+            let txs = block.transactions.iter().skip(1);
+            for t in txs {
+                if t.is_coinbase().is_some() {
+                    return Err("coinbase tx found not in the first position".into());
+                }
+            }
+        }
```

Comment:
Suggestion: I think this can be simplified to:
```suggestion
        if block.transactions.iter().skip(1).any(|tx| tx.contains_coinbase_input()) {
            Err("coinbase tx found not in the first position")?
        }
```

Reply:
Applied this.